native/sort.c

```diff
@@ -0,0 +1,74 @@
+#include <stdio.h>
+
+#include "native.h"
+
+typedef struct MapPair{
+    GoString  key;
+    void*     value;
+} MapPair;
+
+// swap elems in MapPair
+void swap(MapPair* lhs, MapPair* rhs) {
+    MapPair temp;
+    temp = *lhs;
+    *lhs = *rhs;
+    *rhs = temp;
+}
+
+int _strcmp(const char *p,const char *q){
+    while(*p && *q && *p == *q) {
+        p++, q++;
+    }
+
+    return *p - *q;
+}
+
+bool compare(MapPair lhs, MapPair rhs) {
```

Comment:
"compare" name is confuse, maybe "less(lhs, rhs)" is more concrete

Reply:
Also, this function is short, and we can add "inline" to optimize the function call.

Reply:
Thanks, I will make the necessary modifications as suggested.

Reply:
thanks~